api/app/GraphQL/Mutations/SubmitApplication.php

```diff
@@ -114,7 +116,9 @@ public function __invoke($_, array $args)
         $application->pool_candidate_status = ApiEnums::CANDIDATE_STATUS_NEW_APPLICATION;
         $application->expiry_date = $expiryDate;
         $application->signature = $signature;
-        $application->save();
+        $success = $application->save();
```

Comment:
what will happen if there is an error returned from the `save()` ?

Reply:
It depends on the error. If it is a mysql error, an exception is thrown and we should be catching it. If there is any other sort of error, it will simply return `false`. I added this so we only dispatch the event that stores the snapshot if the save is successful with `ApplicationSubmitted::dispatchIf($success, $application);`.